editor/src/clj/editor/ui.clj

```diff
@@ -759,27 +768,24 @@
   HasAction
   (on-action! [node update-fn]
     (.setOnAction node (event-handler e
-                         ;; Clear the auto-commit flag. Further edits will re-apply it.
-                         (when (user-data node ::auto-commit)
-                           (user-data! node ::auto-commit false))
+                         (clear-auto-commit! node)
+                         (update-fn e)
```

Comment:
Use shared function to clear auto-commit flag.

Move `update-fn` call to before we adjust the selection. Not 100% sure of the correct location of the call, but now it is consistent with `TextArea`. I would think it shouldn't matter?

Reply:
I think so as well